x-pack/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -75,19 +75,7 @@ export function initAuthenticateApi(server) {
     method: 'GET',
     path: '/api/security/v1/oidc',
     config: {
-      auth: false,
-      validate: {
-        query: Joi.object().keys({
-          iss: Joi.string(),
-          login_hint: Joi.string(),
-          target_link_uri: Joi.string(),
-          code: Joi.string(),
-          error: Joi.string(),
-          error_description: Joi.string(),
-          error_uri: Joi.string(),
-          state: Joi.string()
-        })
-      }
+      auth: false
```

Comment:
__question:__ can we just use [`Joi.object().keys({...}).unknown()`](https://github.com/hapijs/joi/blob/v13.7.0/API.md#objectunknownallow)?

__question:__ is there any reason why we can't merge these two routes (GET and POST) into one with `method: ['GET', 'POST']`

Reply:
> can we just use Joi.object().keys({...}).unknown()?

Not sure. I'd have to try this out. Are you thinking that we could at least fail early (in validation) when a request reaches kibana with no parameters ? Sounds like a good idea.

> question: is there any reason why we can't merge these two routes (GET and POST) into one with method: ['GET', 'POST']

Good point. Previously we had different validation that's why I had two routes, but now we could merge them

Reply:
> Not sure. I'd have to try this out. Are you thinking that we could at least fail early (in validation) when a request reaches kibana with no parameters ? Sounds like a good idea.

Not exactly, to forbid empty object (when we don't have any query string parameters), I believe we need to add `Joi.object().....min(1)`. My idea was more to have a basic validation for keys where feasible, e.g. `Joi.string().uri()` for `target_link_uri` and `error_uri` or `Joi.string().alphanum().min(1)` for `iss`... or `string().required()` if we have a required field or something like this.

Reply:
Gotcha. I added validation for the URIs and specific for the issuer but I don't think we can add any for code, state.